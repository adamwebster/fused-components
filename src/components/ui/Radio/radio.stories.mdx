import { Meta, Story, Props, Preview } from '@storybook/addon-docs/blocks';
import { withKnobs, select, text, boolean } from '@storybook/addon-knobs';
import { action, actions } from "@storybook/addon-actions";
import { Radio } from './';
import { RadioDemo, RadioDarkMode } from '../../../demos/RadioDemo';
import { DependenciesTable, Dependencies, Dependents } from 'storybook-addon-deps/blocks';

<Meta title="Components / Radio" component={Radio} parameters={{ component: Radio }}/>

# Radio

## An Checkbox component 

<Preview>
      <Radio>Radio</Radio>
       <Radio checked>Radio</Radio>
           <Radio inError>Radio</Radio>
       <Radio inWarning>Radio</Radio>
</Preview>


## Demo 

<Preview withSource={'none'}>
      <RadioDemo/>
</Preview>

## Dark Mode

<RadioDarkMode />

## Properties

<Props of={Radio} />

# Radio Button dependents & dependencies
<DependenciesTable titleDependencies='Dependencies' titleDependents='Dependents' of={Radio} />

