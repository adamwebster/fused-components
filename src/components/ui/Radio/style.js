import styled, { css } from 'styled-components';
import { color } from '../../../styles/styles';

export const Label = styled.span`
    margin-left: 5px;
    display:inline-block;
    ${props => props.inError && css`
        color: ${color.red};
    `}
    ${props => props.inWarning && css`
        color: ${color.yellow};
    `}
`

export const IconStyled = styled.span`
    color: ${color.primary};
    font-size: 15px;
    svg{
       width: 15px;
       top:2px;
       position:relative;
   }
    ${props => props.inError && css`
        color: ${color.red};
    `}
    ${props => props.inWarning && css`
        color: ${color.yellow};
    `}
`

export const RadioInput = styled.input`
position: absolute;
opacity: 0;
cursor: pointer;
`