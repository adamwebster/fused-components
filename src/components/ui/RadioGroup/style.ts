import styled, { css } from 'styled-components';

interface RWProps {
  inline?: boolean;
}
export const RadioWrapper = styled.div<RWProps>`
  width: ${({ inline }) => (inline ? 'fit-content' : '100%')};
  ${({ inline }) =>
    inline &&
    css`
      display: inline-block;
      margin-right: 10px;
    `}
`;
