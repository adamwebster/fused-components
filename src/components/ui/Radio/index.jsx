import React from "react";
import { Label, Icon, RadioInput } from "./style";
import PropTypes from "prop-types";

export const Radio = ({ children, checked, inWarning, inError, ...radioProps }) => {
  return (
    <label>
      <RadioInput inError={inError} inWarning={inWarning} type="radio" {...radioProps} />
      {checked ? <Icon inError={inError} inWarning={inWarning} transform="shrink-10" icon={["fas", "circle"]} mask={['fas', 'circle']}/> : <Icon inError={inError} inWarning={inWarning} icon={["far", "circle"]} />}
      <Label inError={inError} inWarning={inWarning}>{children}</Label>
    </label>
  );
};

Radio.propTypes = {
  checked: PropTypes.bool,
  inError: PropTypes.bool,
  inWarning: PropTypes.bool,
};

Radio.defaultProps = {
  checked: false,
  inError: false,
  inWarning: false,
};
