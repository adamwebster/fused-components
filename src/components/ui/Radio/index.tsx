import React, { ReactNode, ReactElement } from 'react';
import { Label, IconStyled, RadioInput } from './style';
import { Icon } from '../../icon';
import { FCThemeConsumer } from '../../../theming/FCTheme';

export interface Props extends React.HTMLProps<HTMLInputElement> {
  children: ReactNode;
  /** If the radio is checked */
  checked?: boolean;
  /** If the radio is in its warning state */
  inWarning?: boolean;
  /** If the radio is in its error state */
  inError?: boolean;
  /** The function that is called when the input changes between selected and not selected */
  onChange?: (e: React.ChangeEvent<HTMLInputElement>) => void;
  /** The value of the input */
  value?: string;
  id: string;
}
export const Radio = ({
  children,
  checked = false,
  inWarning = false,
  inError = false,
  value,
  onChange = (): void => undefined,
  id,
}: Props): ReactElement => {
  return (
    <FCThemeConsumer>
      {(themeContext): ReactElement => (
        <label>
          <RadioInput
            id={id}
            value={value}
            type="radio"
            checked={checked}
            onChange={(e: React.ChangeEvent<HTMLInputElement>): void => onChange(e)}
          />

          <IconStyled inError={inError} inWarning={inWarning}>
            <Icon aria-labelledby={`labelfor-${id}`} icon={checked ? 'radio-checked' : 'radio'} />
          </IconStyled>

          <Label id={`labelfor-${id}`} theme={themeContext.theme} inError={inError} inWarning={inWarning}>
            {children}
          </Label>
        </label>
      )}
    </FCThemeConsumer>
  );
};
