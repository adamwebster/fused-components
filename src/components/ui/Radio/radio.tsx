import React, { ReactNode, ReactElement, useRef } from 'react';
import { Label, IconStyled, RadioInput } from './style';
import { Icon } from '../../icon';
import { FCThemeConsumer } from '../../../theming/FCTheme';

interface Props extends React.HTMLProps<HTMLInputElement> {
  /** The id of the element. Required for accessibility.  */
  id: string;
  /**
   * @ignore
   */
  children: ReactNode;
  /** If the radio is checked */
  checked?: boolean;
  /** If the radio is in its warning state */
  inWarning?: boolean;
  /** If the radio is in its error state */
  inError?: boolean;
  /** The function that is called when the input changes between selected and not selected */
  onChange?: (e: React.ChangeEvent<HTMLInputElement>) => void;
  /** The value of the input */
  value?: string;
  /**
   * @ignore
   */
  onKeyDown?: (e: any) => void;
  /**
   * @ignore
   */
  onClick?: () => void;
  /**
   * @ignore
   */
  radioTabIndex?: number;
}
const Radio = React.forwardRef<HTMLSpanElement, Props>(
  (
    {
      children,
      checked = false,
      inWarning = false,
      inError = false,
      value,
      onKeyDown = (): void => undefined,
      onChange = (): void => undefined,
      onClick,
      id,
      radioTabIndex = 0,
    }: Props,
    ref,
  ): ReactElement => {
    const RadioRef = useRef(null);
    return (
      <FCThemeConsumer>
        {(themeContext): ReactElement => (
          <label onClick={() => onClick && onClick()}>
            <RadioInput
              id={id}
              value={value}
              type="radio"
              ref={RadioRef}
              tabIndex={-1}
              checked={checked}
              onChange={(e: React.ChangeEvent<HTMLInputElement>): void => onChange(e)}
            />

            <IconStyled
              tabIndex={radioTabIndex}
              ref={ref}
              onKeyDown={e => onKeyDown && onKeyDown(e)}
              role="radio"
              inError={inError}
              inWarning={inWarning}
              theme={themeContext.theme}
              aria-checked={checked ? true : false}
            >
              <Icon aria-labelledby={`labelfor-${id}`} icon={checked ? 'radio-checked' : 'radio'} />
            </IconStyled>

            <Label id={`labelfor-${id}`} theme={themeContext.theme} inError={inError} inWarning={inWarning}>
              {children}
            </Label>
          </label>
        )}
      </FCThemeConsumer>
    );
  },
);

Radio.displayName = 'Radio';

export default Radio;
