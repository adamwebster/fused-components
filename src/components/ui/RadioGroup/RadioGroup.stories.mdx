import {useState} from 'react';
import { Meta, Story, ArgsTable, Canvas } from '@storybook/addon-docs/blocks';
import { action, actions } from "@storybook/addon-actions";
import { DependenciesTable, Dependencies, Dependents } from 'storybook-addon-deps/blocks';
import { DarkModeWrapper } from "../../../common/styles";
import { color } from '../../../styles/styles';
import { FCThemeProvider } from "../../../theming/FCTheme";
import { RadioGroup } from './';
import { Radio } from '../Radio';

<Meta title="Components / Radio Group" component={RadioGroup} parameters={{ component: RadioGroup }}/>

# Radio Group

The radio group wraps multiple radios and adds functionally that is not available by default with the basic radio component such as keyboard support and only allowing one option to be selected at once. 

## Default
<Canvas>
    <Story name="Radio Group" parameters={{ decorators: []}}>
        <RadioGroup selectedValue='radio 2' onRadioChanged={(value) => console.log(value)}>
            <Radio id="1" value="radio 1">Radio 1</Radio>
            <Radio id="2" value="radio 2">Radio 2</Radio>
            <Radio id="3" value="radio 3">Radio 3</Radio>
        </RadioGroup>
    </Story>
</Canvas>

<Canvas>
    <Story name="Radio Group Inline" parameters={{ decorators: []}}>
        <RadioGroup inline onRadioChanged={(value) => console.log(value)}>
            <Radio id="1" value="radio 1">Radio 1</Radio>
            <Radio id="2" value="radio 2">Radio 2</Radio>
            <Radio id="3" value="radio 3">Radio 3</Radio>
        </RadioGroup>
    </Story>
</Canvas>

<Story name="Radio Group Dark Mode"
decorators={[storyFn => <div style={{ backgroundColor: color.darkModeBG, height: 'calc(' + 100 + 'vh - ' + 20 + 'px)' , boxSizing: 'border-box', padding: 40 + 'px' }}>{storyFn()}</div>]}
parameters={{ docs: { disable: true  }}}
>
    <FCThemeProvider value={{theme: 'dark'}}>
        <RadioGroup selectedValue='radio 2' onRadioChanged={(value) => console.log(value)}>
            <Radio id="1" value="radio 1">Radio 1</Radio>
            <Radio id="2" value="radio 2">Radio 2</Radio>
            <Radio id="3" value="radio 3">Radio 3</Radio>
        </RadioGroup>
    </FCThemeProvider>
</Story>


## Properties

<ArgsTable for={RadioGroup} />
