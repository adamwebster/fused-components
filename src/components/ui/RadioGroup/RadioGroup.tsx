import React, { ReactNode, ReactElement, useState, useEffect } from 'react';
import { Radio } from '../Radio';
import { RadioWrapper } from './style';
interface Props {
  children: ReactNode;
  onChange?: (value: string) => void;
  inline?: boolean;
  value?: string;
}

const RadioGroup = ({ children, onChange = (): void => undefined, inline, value }: Props): ReactElement => {
  const [radioItems, setRadioItems] = useState<Array<any>>([]);
  const [checkedItem, setCheckedItem] = useState(0);
  const radioRefs: any = [];

  const increaseSelectedIndex = (e: any): void => {
    e.preventDefault();
    if (checkedItem + 1 !== radioRefs.length) {
      setCheckedItem(checkedItem + 1);
      radioRefs[checkedItem + 1].focus();
      onChange(radioItems[checkedItem + 1].props.value);
    } else {
      setCheckedItem(0);
      radioRefs[0].focus();
      onChange(radioItems[0].props.value);
    }
  };

  const decreaseSelectedIndex = (e: any): void => {
    e.preventDefault();
    if (checkedItem !== 0) {
      setCheckedItem(checkedItem - 1);
      radioRefs[checkedItem - 1].focus();
      onChange(radioItems[checkedItem - 1].props.value);
    } else {
      setCheckedItem(radioRefs.length - 1);
      radioRefs[radioRefs.length - 1].focus();
      onChange(radioItems[radioRefs.length - 1].props.value);
    }
  };

  const handleKeyDown = (e: any, index: number): void => {
    if (e.key === 'ArrowRight' || e.key === 'ArrowDown') {
      increaseSelectedIndex(e);
    }
    if (e.key === 'ArrowLeft' || e.key == 'ArrowUp') {
      decreaseSelectedIndex(e);
    }
    if (e.key === ' ') {
      e.preventDefault();
      setCheckedItem(index);
      onChange(radioItems[index].props.value);
    }
  };

  useEffect(() => {
    const ArrayToSave: any = [];
    React.Children.map(children, (child: any, index: number): void => {
      if (child.type === Radio) {
        ArrayToSave.push(child);
      }
      if (child.props.value === value) {
        setCheckedItem(index);
      }
    });
    setRadioItems(ArrayToSave);
  }, [children]);

  return (
    <div>
      {radioItems.map((radio: any, index) => {
        return (
          <RadioWrapper inline={inline} key={radio.props.id}>
            <Radio
              onKeyDown={(e: any): void => handleKeyDown(e, index)}
              checked={checkedItem === index || false}
              id={radio.props.key || radio.props.id}
              ref={(ref: any): void => radioRefs.push(ref)}
              value={radio.props.value}
              radioTabIndex={0}
              onClick={(): void => {
                onChange(radio.props.value);
                setCheckedItem(index);
                radioRefs[index].focus();
              }}
            >
              {radio.props.children}
            </Radio>
          </RadioWrapper>
        );
      })}
    </div>
  );
};

export default RadioGroup;
