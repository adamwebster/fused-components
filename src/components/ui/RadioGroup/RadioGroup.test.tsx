import React from 'react';
import { render, screen, fireEvent } from '@testing-library/react';
import 'jest-styled-components';
import { Radio } from '../Radio/index';
import RadioGroup from './RadioGroup';
import userEvent from '@testing-library/user-event';

describe('Radio Group Test', () => {
  test('Renders the radio group component', () => {
    const { container } = render(
      <RadioGroup>
        <Radio id="1" value="Radio 1">
          Radio 1
        </Radio>
        <Radio id="2" value="Radio 2">
          Radio 2
        </Radio>
        <Radio id="3" value="Radio 3">
          Radio 3
        </Radio>
      </RadioGroup>,
    );
    expect(container).toMatchSnapshot();
  });

  test('Renders the radio group component inline', () => {
    const { container } = render(
      <RadioGroup inline>
        <Radio id="1" value="Radio 1">
          Radio 1
        </Radio>
        <Radio id="2" value="Radio 2">
          Radio 2
        </Radio>
        <Radio id="3" value="Radio 3">
          Radio 3
        </Radio>
      </RadioGroup>,
    );
    expect(container).toMatchSnapshot();
  });
  test('Clicking on a radio changes the value', () => {
    let value = 'Hello';

    const onChange = jest.fn(rv => {
      value = rv;
    });
    render(
      <RadioGroup onChange={value => onChange(value)}>
        <Radio id="1" value="Radio 1">
          Radio 1
        </Radio>
        <Radio id="2" value="Radio 2">
          Radio 2
        </Radio>
        <Radio id="3" value="Radio 3">
          Radio 3
        </Radio>
      </RadioGroup>,
    );
    const secondRadio = screen.getByText('Radio 2');
    userEvent.click(secondRadio);
    expect(value).toBe('Radio 2');
  });

  test('Clicking down when a radio is focused changes the value', () => {
    let value = 'Hello';

    const onChange = jest.fn(rv => {
      value = rv;
    });
    render(
      <RadioGroup onChange={value => onChange(value)}>
        <Radio id="1" value="Radio 1">
          Radio 1
        </Radio>
        <Radio id="2" value="Radio 2">
          Radio 2
        </Radio>
        <Radio id="3" value="Radio 3">
          Radio 3
        </Radio>
      </RadioGroup>,
    );
    userEvent.tab();
    const firstRadio = screen.getAllByRole('radio')[1];
    expect(firstRadio).toHaveFocus();
    fireEvent.keyDown(firstRadio, { key: 'ArrowDown' });

    expect(onChange).toHaveBeenCalledTimes(1);
    expect(value).toBe('Radio 2');
  });

  test('Clicking right when a radio is focused changes the value', () => {
    let value = 'Hello';

    const onChange = jest.fn(rv => {
      value = rv;
    });
    render(
      <RadioGroup onChange={value => onChange(value)}>
        <Radio id="1" value="Radio 1">
          Radio 1
        </Radio>
        <Radio id="2" value="Radio 2">
          Radio 2
        </Radio>
        <Radio id="3" value="Radio 3">
          Radio 3
        </Radio>
      </RadioGroup>,
    );
    userEvent.tab();
    const firstRadio = screen.getAllByRole('radio')[1];
    expect(firstRadio).toHaveFocus();
    fireEvent.keyDown(firstRadio, { key: 'ArrowRight' });

    expect(onChange).toHaveBeenCalledTimes(1);
    expect(value).toBe('Radio 2');
  });

  test('Clicking up when a radio is focused changes the value', () => {
    let value = 'Hello';

    const onChange = jest.fn(rv => {
      value = rv;
    });
    render(
      <RadioGroup onChange={value => onChange(value)}>
        <Radio id="1" value="Radio 1">
          Radio 1
        </Radio>
        <Radio id="2" value="Radio 2">
          Radio 2
        </Radio>
        <Radio id="3" value="Radio 3">
          Radio 3
        </Radio>
      </RadioGroup>,
    );
    userEvent.tab();
    const firstRadio = screen.getAllByRole('radio')[1];
    expect(firstRadio).toHaveFocus();
    fireEvent.keyDown(firstRadio, { key: 'ArrowDown' });
    fireEvent.keyDown(firstRadio, { key: 'ArrowUp' });

    expect(onChange).toHaveBeenCalledTimes(2);
    expect(value).toBe('Radio 1');
  });

  test('Clicking left when a radio is focused changes the value', () => {
    let value = 'Hello';

    const onChange = jest.fn(rv => {
      value = rv;
    });
    render(
      <RadioGroup onChange={value => onChange(value)}>
        <Radio id="1" value="Radio 1">
          Radio 1
        </Radio>
        <Radio id="2" value="Radio 2">
          Radio 2
        </Radio>
        <Radio id="3" value="Radio 3">
          Radio 3
        </Radio>
      </RadioGroup>,
    );
    userEvent.tab();
    const firstRadio = screen.getAllByRole('radio')[1];
    expect(firstRadio).toHaveFocus();
    fireEvent.keyDown(firstRadio, { key: 'ArrowRight' });
    fireEvent.keyDown(firstRadio, { key: 'ArrowLeft' });

    expect(onChange).toHaveBeenCalledTimes(2);
    expect(value).toBe('Radio 1');
  });

  test('Clicking space on a focused radio changes the value', () => {
    let value = 'Hello';

    const onChange = jest.fn(rv => {
      value = rv;
    });
    render(
      <RadioGroup onChange={value => onChange(value)}>
        <Radio id="1" value="Radio 1">
          Radio 1
        </Radio>
        <Radio id="2" value="Radio 2">
          Radio 2
        </Radio>
        <Radio id="3" value="Radio 3">
          Radio 3
        </Radio>
      </RadioGroup>,
    );
    const secondRadio = screen.getAllByRole('radio')[3];
    secondRadio.focus();
    expect(secondRadio).toHaveFocus();
    fireEvent.keyDown(secondRadio, { key: ' ' });

    expect(onChange).toHaveBeenCalledTimes(1);
    expect(value).toBe('Radio 2');
  });

  test('Clicking down when the last radio is focused sets the value to the first radios value', () => {
    let value = 'Hello';

    const onChange = jest.fn(rv => {
      value = rv;
    });
    render(
      <RadioGroup onChange={value => onChange(value)}>
        <Radio id="1" value="Radio 1">
          Radio 1
        </Radio>
        <Radio id="2" value="Radio 2">
          Radio 2
        </Radio>
        <Radio id="3" value="Radio 3">
          Radio 3
        </Radio>
      </RadioGroup>,
    );
    const lastRadio = screen.getAllByRole('radio')[5];
    lastRadio.focus();
    fireEvent.keyDown(lastRadio, { key: ' ' });
    expect(lastRadio).toHaveFocus();
    fireEvent.keyDown(lastRadio, { key: 'ArrowDown' });

    expect(onChange).toHaveBeenCalledTimes(2);
    expect(value).toBe('Radio 1');
  });

  test('Clicking up when the first radio is focused sets the value to the last radios value', () => {
    let value = 'Hello';

    const onChange = jest.fn(rv => {
      value = rv;
    });
    render(
      <RadioGroup onChange={value => onChange(value)}>
        <Radio id="1" value="Radio 1">
          Radio 1
        </Radio>
        <Radio id="2" value="Radio 2">
          Radio 2
        </Radio>
        <Radio id="3" value="Radio 3">
          Radio 3
        </Radio>
      </RadioGroup>,
    );
    const lastRadio = screen.getAllByRole('radio')[1];
    lastRadio.focus();
    fireEvent.keyDown(lastRadio, { key: ' ' });
    expect(lastRadio).toHaveFocus();
    fireEvent.keyDown(lastRadio, { key: 'ArrowUp' });

    expect(onChange).toHaveBeenCalledTimes(2);
    expect(value).toBe('Radio 3');
  });

  test('Setting the value of the radio group changes the selected radio', () => {
    render(
      <RadioGroup selectedValue="Radio 2">
        <Radio id="1" value="Radio 1">
          Radio 1
        </Radio>
        <Radio id="2" value="Radio 2">
          Radio 2
        </Radio>
        <Radio id="3" value="Radio 3">
          Radio 3
        </Radio>
      </RadioGroup>,
    );
    const secondRadio = screen.getAllByRole('radio')[3];
    expect(secondRadio.getAttribute('aria-checked')).toBe('true');
  });
});
