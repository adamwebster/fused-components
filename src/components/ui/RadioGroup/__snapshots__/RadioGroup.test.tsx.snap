// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Radio Group Test Renders the radio group component 1`] = `
.c4 {
  margin-left: 5px;
  display: inline-block;
}

.c2 {
  color: #006EF5;
  font-size: 15px;
}

.c2 svg {
  width: 15px;
  top: 2px;
  position: relative;
}

.c1 {
  position: absolute;
  opacity: 0;
  cursor: pointer;
}

.c3 {
  width: 1em;
  height: 1em;
}

.c0 {
  width: 100%;
}

<div>
  <div
    class="c0"
  >
    <label>
      <input
        checked=""
        class="c1"
        id="1"
        tabindex="-1"
        type="radio"
        value="Radio 1"
      />
      <span
        aria-checked="true"
        class="c2"
        role="radio"
        tabindex="0"
      >
        <svg
          aria-labelledby="labelfor-1"
          class="c3  radio-checked"
          clip-rule="evenodd"
          fill-rule="evenodd"
          role="img"
          stroke-linejoin="round"
          stroke-miterlimit="2"
          viewBox="0 0 48 48"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M24,0c13.246,0 24,10.754 24,24c0,13.246 -10.754,24 -24,24c-13.246,0 -24,-10.754 -24,-24c0,-13.246 10.754,-24 24,-24Zm0,8c8.831,0 16,7.169 16,16c0,8.831 -7.169,16 -16,16c-8.831,0 -16,-7.169 -16,-16c0,-8.831 7.169,-16 16,-16Zm0,8c4.415,0 8,3.585 8,8c0,4.415 -3.585,8 -8,8c-4.415,0 -8,-3.585 -8,-8c0,-4.415 3.585,-8 8,-8Z"
            fill="currentColor"
          />
        </svg>
      </span>
      <span
        class="c4"
        id="labelfor-1"
      >
        Radio 1
      </span>
    </label>
  </div>
  <div
    class="c0"
  >
    <label>
      <input
        class="c1"
        id="2"
        tabindex="-1"
        type="radio"
        value="Radio 2"
      />
      <span
        aria-checked="false"
        class="c2"
        role="radio"
        tabindex="0"
      >
        <svg
          aria-labelledby="labelfor-2"
          class="c3  radio"
          clip-rule="evenodd"
          fill-rule="evenodd"
          role="img"
          stroke-linejoin="round"
          stroke-miterlimit="2"
          viewBox="0 0 48 48"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M24,0c13.246,0 24,10.754 24,24c0,13.246 -10.754,24 -24,24c-13.246,0 -24,-10.754 -24,-24c0,-13.246 10.754,-24 24,-24Zm0,8c8.831,0 16,7.169 16,16c0,8.831 -7.169,16 -16,16c-8.831,0 -16,-7.169 -16,-16c0,-8.831 7.169,-16 16,-16Z"
            fill="currentColor"
          />
        </svg>
      </span>
      <span
        class="c4"
        id="labelfor-2"
      >
        Radio 2
      </span>
    </label>
  </div>
  <div
    class="c0"
  >
    <label>
      <input
        class="c1"
        id="3"
        tabindex="-1"
        type="radio"
        value="Radio 3"
      />
      <span
        aria-checked="false"
        class="c2"
        role="radio"
        tabindex="0"
      >
        <svg
          aria-labelledby="labelfor-3"
          class="c3  radio"
          clip-rule="evenodd"
          fill-rule="evenodd"
          role="img"
          stroke-linejoin="round"
          stroke-miterlimit="2"
          viewBox="0 0 48 48"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M24,0c13.246,0 24,10.754 24,24c0,13.246 -10.754,24 -24,24c-13.246,0 -24,-10.754 -24,-24c0,-13.246 10.754,-24 24,-24Zm0,8c8.831,0 16,7.169 16,16c0,8.831 -7.169,16 -16,16c-8.831,0 -16,-7.169 -16,-16c0,-8.831 7.169,-16 16,-16Z"
            fill="currentColor"
          />
        </svg>
      </span>
      <span
        class="c4"
        id="labelfor-3"
      >
        Radio 3
      </span>
    </label>
  </div>
</div>
`;

exports[`Radio Group Test Renders the radio group component inline 1`] = `
.c4 {
  margin-left: 5px;
  display: inline-block;
}

.c2 {
  color: #006EF5;
  font-size: 15px;
}

.c2 svg {
  width: 15px;
  top: 2px;
  position: relative;
}

.c1 {
  position: absolute;
  opacity: 0;
  cursor: pointer;
}

.c3 {
  width: 1em;
  height: 1em;
}

.c0 {
  width: -webkit-fit-content;
  width: -moz-fit-content;
  width: fit-content;
  display: inline-block;
  margin-right: 10px;
}

<div>
  <div
    class="c0"
  >
    <label>
      <input
        checked=""
        class="c1"
        id="1"
        tabindex="-1"
        type="radio"
        value="Radio 1"
      />
      <span
        aria-checked="true"
        class="c2"
        role="radio"
        tabindex="0"
      >
        <svg
          aria-labelledby="labelfor-1"
          class="c3  radio-checked"
          clip-rule="evenodd"
          fill-rule="evenodd"
          role="img"
          stroke-linejoin="round"
          stroke-miterlimit="2"
          viewBox="0 0 48 48"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M24,0c13.246,0 24,10.754 24,24c0,13.246 -10.754,24 -24,24c-13.246,0 -24,-10.754 -24,-24c0,-13.246 10.754,-24 24,-24Zm0,8c8.831,0 16,7.169 16,16c0,8.831 -7.169,16 -16,16c-8.831,0 -16,-7.169 -16,-16c0,-8.831 7.169,-16 16,-16Zm0,8c4.415,0 8,3.585 8,8c0,4.415 -3.585,8 -8,8c-4.415,0 -8,-3.585 -8,-8c0,-4.415 3.585,-8 8,-8Z"
            fill="currentColor"
          />
        </svg>
      </span>
      <span
        class="c4"
        id="labelfor-1"
      >
        Radio 1
      </span>
    </label>
  </div>
  <div
    class="c0"
  >
    <label>
      <input
        class="c1"
        id="2"
        tabindex="-1"
        type="radio"
        value="Radio 2"
      />
      <span
        aria-checked="false"
        class="c2"
        role="radio"
        tabindex="0"
      >
        <svg
          aria-labelledby="labelfor-2"
          class="c3  radio"
          clip-rule="evenodd"
          fill-rule="evenodd"
          role="img"
          stroke-linejoin="round"
          stroke-miterlimit="2"
          viewBox="0 0 48 48"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M24,0c13.246,0 24,10.754 24,24c0,13.246 -10.754,24 -24,24c-13.246,0 -24,-10.754 -24,-24c0,-13.246 10.754,-24 24,-24Zm0,8c8.831,0 16,7.169 16,16c0,8.831 -7.169,16 -16,16c-8.831,0 -16,-7.169 -16,-16c0,-8.831 7.169,-16 16,-16Z"
            fill="currentColor"
          />
        </svg>
      </span>
      <span
        class="c4"
        id="labelfor-2"
      >
        Radio 2
      </span>
    </label>
  </div>
  <div
    class="c0"
  >
    <label>
      <input
        class="c1"
        id="3"
        tabindex="-1"
        type="radio"
        value="Radio 3"
      />
      <span
        aria-checked="false"
        class="c2"
        role="radio"
        tabindex="0"
      >
        <svg
          aria-labelledby="labelfor-3"
          class="c3  radio"
          clip-rule="evenodd"
          fill-rule="evenodd"
          role="img"
          stroke-linejoin="round"
          stroke-miterlimit="2"
          viewBox="0 0 48 48"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M24,0c13.246,0 24,10.754 24,24c0,13.246 -10.754,24 -24,24c-13.246,0 -24,-10.754 -24,-24c0,-13.246 10.754,-24 24,-24Zm0,8c8.831,0 16,7.169 16,16c0,8.831 -7.169,16 -16,16c-8.831,0 -16,-7.169 -16,-16c0,-8.831 7.169,-16 16,-16Z"
            fill="currentColor"
          />
        </svg>
      </span>
      <span
        class="c4"
        id="labelfor-3"
      >
        Radio 3
      </span>
    </label>
  </div>
</div>
`;
